numba/_lapack.c

```diff
@@ -0,0 +1,1035 @@
+/*
+ * This file contains wrappers of BLAS and LAPACK functions
+ */
+#include <complex.h>
+
+/*
+ * BLAS calling helpers.  The helpers can be called without the GIL held.
+ * The caller is responsible for checking arguments (especially dimensions).
+ */
+
+/* Fast getters caching the value of a function's address after
+   the first call to import_cblas_function(). */
+
+#define EMIT_GET_CBLAS_FUNC(name)                                 \
+    static void *cblas_ ## name = NULL;                           \
+    static void *get_cblas_ ## name(void) {                       \
+        if (cblas_ ## name == NULL) {                             \
+            PyGILState_STATE st = PyGILState_Ensure();            \
+            const char *mod = "scipy.linalg.cython_blas";         \
+            cblas_ ## name = import_cython_function(mod, # name); \
+            PyGILState_Release(st);                               \
+        }                                                         \
+        return cblas_ ## name;                                    \
+    }
+
+EMIT_GET_CBLAS_FUNC(dgemm)
+EMIT_GET_CBLAS_FUNC(sgemm)
+EMIT_GET_CBLAS_FUNC(cgemm)
+EMIT_GET_CBLAS_FUNC(zgemm)
+EMIT_GET_CBLAS_FUNC(dgemv)
+EMIT_GET_CBLAS_FUNC(sgemv)
+EMIT_GET_CBLAS_FUNC(cgemv)
+EMIT_GET_CBLAS_FUNC(zgemv)
+EMIT_GET_CBLAS_FUNC(ddot)
+EMIT_GET_CBLAS_FUNC(sdot)
+EMIT_GET_CBLAS_FUNC(cdotu)
+EMIT_GET_CBLAS_FUNC(zdotu)
+EMIT_GET_CBLAS_FUNC(cdotc)
+EMIT_GET_CBLAS_FUNC(zdotc)
+
+#undef EMIT_GET_CBLAS_FUNC
+
+/*
+ * Define what a Fortran "int" is, some LAPACKs have 64 bit integer support
```

Comment:
Everything in the Python ecosystem (and also in our conda packages) I'm aware of is using the default ABI on each platform.